src/common/paddle/organization/processing.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  SubscriptionCreatedEvent,
+  SubscriptionCreatedNotification,
+} from '@paddle/paddle-node-sdk';
+import { randomUUID } from 'crypto';
+import createOrGetConnection from '../../../db';
+import { extractSubscriptionCycle, paddleInstance } from '..';
+import { Organization, User } from '../../../entity';
+import { ContentPreferenceOrganization } from '../../../entity/contentPreference/ContentPreferenceOrganization';
+import { ContentPreferenceStatus } from '../../../entity/contentPreference/types';
+import { OrganizationMemberRoles } from '../../../roles';
+import {
+  SubscriptionProcessor,
+  SubscriptionProvider,
+  SubscriptionStatus,
+} from '../../plus';
+import { logger } from '../../../logger';
+import { isPlusMember } from '../../../paddle';
+import { updateSubscriptionFlags } from '../../utils';
+
+type PaddleCustomData = {
+  user_id?: string;
+};
+
+// If the user provides business information during the checkout, we will use it
+// otherwise we will use the default business name
+const getBusinessName = async (
+  user: User,
+  data: SubscriptionCreatedNotification,
+) => {
+  const defaultBusinessName = `${user.name}'s new Organization`;
+  if (!data.businessId) {
+    return defaultBusinessName;
+  }
+
+  try {
+    const business = await paddleInstance.businesses.get(
+      data.customerId,
+      data.businessId,
+    );
+    return business.name;
+  } catch (error) {
+    return defaultBusinessName;
+  }
+};
+
+export const createOrganizationSubscription = async ({
+  event,
+}: {
+  event: SubscriptionCreatedEvent;
+}) => {
+  const organizationId = randomUUID();
+  const con = await createOrGetConnection();
+
+  const { data } = event;
+  const { user_id: userId }: PaddleCustomData = data?.customData || {};
```

Comment:
maybe add zod schema like we have for Cores, and pick up what you need from raw event, that way you don't have to do bunch of checks below if data is here, you just use zod type

for example make sure that there is one item in items, quantity exists, customData has user_id.

Reply:
I'll take a look at cores. I just based it on plus 🙈